prow/cmd/autobump/bump.sh

```diff
@@ -66,34 +66,54 @@ main() {
   else
     usage
   fi
-  echo -e "Bumping: 'gcr.io/k8s-prow/' images to $(color-version ${new_version}) ..." >&2
+  echo -e "Bumping: 'gcr.io/k8s-prow/' images to $(color-version "${new_version}") ..." >&2
 
   bumpfiles=($(add_suffix "$(split_on_commas "$COMPONENT_FILE_DIR")"))
   bumpfiles+=("${CONFIG_PATH}")
   if [[ -n "${JOB_CONFIG_PATH}" ]]; then
     bumpfiles+=($(grep -rl -e "gcr.io/k8s-prow/" "${JOB_CONFIG_PATH}"))
   fi
 
-  echo -e "Attempting to bump the following files: ${bumpfiles[@]}" >&2
-
+  echo "Attempting to bump the following files:" >&2
+  for bf in "${bumpfiles[@]}"; do
+    echo -e "$bf"
+  done
+  local token="$(gcloud auth print-access-token)"
   # Update image tags in the identified files.
-  filter="s/gcr.io\/k8s-prow\/\([[:alnum:]_-]\+\):v[a-f0-9-]\+/gcr.io\/k8s-prow\/\1:${new_version}/I"
+  local matcher="gcr.io\/k8s-prow\/\([[:alnum:]_-]\+\):v[a-f0-9-]\+"
+  local replacer="s/${matcher}/gcr.io\/k8s-prow\/\1:${new_version}/I"
   for file in "${bumpfiles[@]}"; do
-    ${SED} -i "${filter}" ${file}
+    local images="$(grep -o "${matcher}" "${file}")"
+    local array=(${images//\\n/})
+    for i in "${!array[@]}"; do
+      # image is in the format of gcr.io/k8s-prow/[image_name]:[tag]
+      local image="${array[i]}"
+      echo "Checking the existence of ${image} "
+      # Use the Docker Registry v2 API to query the image manifest to check if the given image tag exists or not.
+      # The manifest_url is in the format of https://gcr.io/v2/k8s-prow/[image_name]/manifests/[tag]
+      # Check more details from https://stackoverflow.com/a/55344819/13578870
+      local manifest_url=$(echo "$image" | sed "s/:/\/manifests\//" | sed "s/gcr.io/https:\/\/gcr.io\/v2/")
+      local code=$(curl --fail -L -H "Authorization: Bearer $token" -o /dev/null -s -w "%{http_code}" "${manifest_url}")
```

Comment:
IMO there's no need for the `-w %{http_code}` the `--fail` handles this.

Reply:
Thanks, done.